### ISSUE TITLE
Invalid risky-shell-pipe with pwsh shell

### ISSUE THREAD

**Issue by rgl** [ansible/ansible-lint#3161]
##### Summary

ansible-lint 6.4.1 started emitting invalid `risky-shell-pipe: Shells that use pipes should set the pipefail option.` for pwsh:

```
WARNING  Listing 1 violation(s) that are fatal
roles/vmware_client/tasks/powercli.yml:1: risky-shell-pipe: Shells that use pipes should set the pipefail option.
Read documentation for instructions on how to ignore specific rule violations.

               Rule Violation Summary                
 count tag              profile rule associated tags 
     1 risky-shell-pipe safety  command-shell        
```

For the snippet:

```yaml
- name: Install powercli
  ansible.builtin.shell:
    executable: /bin/pwsh
    cmd: |
      Set-StrictMode -Version Latest
      $ProgressPreference = 'SilentlyContinue'
      $ErrorActionPreference = 'Stop'
      trap {
        Write-Output "ERROR: $_"
        Write-Output (($_.ScriptStackTrace -split '\r?\n') -replace '^(.*)$','ERROR: $1')
        Write-Output (($_.Exception.ToString() -split '\r?\n') -replace '^(.*)$','ERROR EXCEPTION: $1')
        Exit 1
      }
...
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
# ansible-lint --version
ansible-lint 6.14.1 using ansible 2.14.3
```

##### STEPS TO REPRODUCE

Execute ansible-lint.

##### Desired Behavior

It should not apply that lint to `pwsh`.
